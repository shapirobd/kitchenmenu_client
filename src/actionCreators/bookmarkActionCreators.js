import axios from "axios";
import { BOOKMARK_RECIPE, UNBOOKMARK_RECIPE } from "../components/actionTypes";

// adds bookmarked recipe to database as well as redux state
// by dispatching action created by bookmarkedRecipe()
export const bookmarkRecipe = (username, recipeId) => {
	return async (dispatch) => {
		try {
			await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/users/bookmarkRecipe`,
				{
					username,
					recipeId,
				}
			);
			dispatch(bookmarkedRecipe(recipeId));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe to be bookmarked
const bookmarkedRecipe = (recipeId) => {
	return {
		type: BOOKMARK_RECIPE,
		payload: {
			recipeId,
		},
	};
};

// removes bookmarked recipe from database as well as redux state
// by dispatching action created by unbookmarkedRecipe()
export const unbookmarkRecipe = (username, recipeId) => {
	return async (dispatch) => {
		try {
			await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/users/unbookmarkRecipe`,
				{
					username,
					recipeId,
				}
			);
			dispatch(unbookmarkedRecipe(recipeId));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe to be unbookmarked
const unbookmarkedRecipe = (recipeId) => {
	return {
		type: UNBOOKMARK_RECIPE,
		payload: {
			recipeId,
		},
	};
};
