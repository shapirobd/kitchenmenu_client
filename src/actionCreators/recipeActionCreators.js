import axios from "axios";
import { LOAD_FEED, LOAD_RECIPE, FILTER_FEED } from "../components/actionTypes";
import createMacrosParams from "../helpers/createMacrosParams";

// retrieves all recipes from spoonacular API (paginated 40 per page) and updates
// the feed & page number in redux state by dispatching array of recipes and the page number
export const loadFeed = (page = 1, filterData = { diets: [], macros: {} }) => {
	return async (dispatch) => {
		try {
			const { diets, macros, recipeName } = filterData;
			const macrosParams = createMacrosParams(macros);
			const dietsParams = diets.join(",");
			const recipes = await axios.get(
				"https://api.spoonacular.com/recipes/complexSearch",
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						offset: 40 * (page - 1),
						number: 40,
						diet: dietsParams,
						...macrosParams,
						query: recipeName,
					},
				}
			);
			dispatch(loadedFeed(recipes.data, page));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action containing recipes to include in redux state's feed,
// total number of results found to set as redux state's totalResults,
// and the current page number to set as redux state's page
const loadedFeed = (data, page) => {
	return {
		type: LOAD_FEED,
		payload: {
			recipes: data.results,
			totalResults: data.totalResults <= 900 ? data.totalResults : 900,
			page,
		},
	};
};

// retrieves data on a single recipe from spoonacular API and updates
// currentRecipe in redux state by dispatching recipe info & instructions info
export const loadRecipe = (recipeId) => {
	console.log("2: ", recipeId);
	return async (dispatch) => {
		try {
			const recipe = await axios.get(
				`https://api.spoonacular.com/recipes/${recipeId}/information`,
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						includeNutrition: true,
					},
				}
			);
			const instructions = await axios.get(
				`https://api.spoonacular.com/recipes/${recipeId}/analyzedInstructions`,
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						stepBreakdown: true,
					},
				}
			);
			dispatch(loadedRecipe(recipe.data, instructions.data[0]));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action containing information to include in redux state's currentRecipe
const loadedRecipe = (recipe, instructions) => {
	return {
		type: LOAD_RECIPE,
		payload: { currentRecipe: { recipe, instructions } },
	};
};

// retrieves all recipes from spoonacular API with given ingredients (pagination not an option)
// and updates feed, totalResults, page, and isIngredientBased in redux state by
// dispatching array of found recipes and the length of that array
export const getRecipesByIngredients = (ingredients) => {
	return async (dispatch) => {
		try {
			const ingredientsParams = ingredients.join(",");
			const recipes = await axios.get(
				"https://api.spoonacular.com/recipes/findByIngredients",
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						offset: 0,
						number: 6900,
						ingredients: ingredientsParams,
					},
				}
			);
			dispatch(
				filteredFeed({ results: recipes.data, totalResults: recipes.length })
			);
		} catch (e) {
			console.error(e);
		}
	};
};

// export const getRecipesByKeywords = (keywords) => {
// 	return async (dispatch) => {
// 		try {
// 			const ingredientsParams = ingredients.join(",");
// 			const recipes = await axios.get(
// 				"https://api.spoonacular.com/recipes/findByIngredients",
// 				{
// 					params: {
// 						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
// 						offset: 0,
// 						number: 6900,
// 						query: keywords,
// 					},
// 				}
// 			);
// 			dispatch(
// 				filteredFeed({ results: recipes.data, totalResults: recipes.length })
// 			);
// 		} catch (e) {
// 			console.error(e);
// 		}
// 	};
// };

// const gotRecipesByIngredients = (results) => {
// 	return {
// 		type: FILTER_BY_INGREDIENTS,
// 		payload: { recipes: data.results, totalResults: data.totalResults },
// 	};
// };

// export const filterFeed = (filterData, page = 1) => {
// 	return async (dispatch) => {
// 		try {
// 			const { diets, macros, recipeName } = filterData;
// 			const macrosParams = createMacrosParams(macros);
// 			const dietsParams = diets.join(",");
// 			const recipes = await axios.get(
// 				"https://api.spoonacular.com/recipes/complexSearch",
// 				{
// 					params: {
// 						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
// 						offset: 40 * (page - 1),
// 						number: 40,
// 						diet: dietsParams,
// 						...macrosParams,
// 						query: recipeName,
// 					},
// 				}
// 			);
// 			console.log(recipes.data);
// 			dispatch(filteredFeed(recipes.data, page));
// 		} catch (e) {
// 			console.error(e);
// 		}
// 	};
// };

// returns action containing information to include in redux state's feed, totalResults, and page
const filteredFeed = (data, page) => {
	return {
		type: FILTER_FEED,
		payload: { recipes: data.results, totalResults: data.totalResults, page },
	};
};
