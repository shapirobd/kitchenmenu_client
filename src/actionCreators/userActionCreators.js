import axios from "axios";
import { LOGIN, LOGOUT } from "../components/actionTypes";

// adds user to database as well as redux state
// by dispatching action created by registered()
export const register = (data) => {
	return async (dispatch) => {
		try {
			const resp = await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/auth/register`,
				data
			);
			// const userResp = await axios.get(`${API_URL}/users/${data.username}`);
			const { user, token } = resp.data;
			dispatch(registered(token, user));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// newly registered user
const registered = (token, user) => {
	return {
		type: LOGIN,
		payload: {
			user,
			token,
		},
	};
};

// verifies that username & password from data are valid
// if valid, adds the user to redux state by dispatching action created by loggedIn()
// in invalid, throws error
export const login = (data) => {
	return async (dispatch) => {
		try {
			const resp = await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/auth/login`,
				data
			);
			const { user, token } = resp.data;
			dispatch(loggedIn(token, user));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// logged in user
const loggedIn = (token, user) => {
	return {
		type: LOGIN,
		payload: {
			user,
			token,
		},
	};
};

// returns action to be dispatched that ends up removing
// the user's information from redux state
export const logout = () => {
	return {
		type: LOGOUT,
	};
};

// edits the information on a given user in the database
// as well as redux state by dispatching action created by edittedProfile()
export const editProfile = (username, data) => {
	return async (dispatch) => {
		try {
			const editResp = await axios.patch(
				`${process.env.API_URL || "http://localhost:5000"}/users/${username}`,
				data
			);
			dispatch(edittedProfile(editResp.data));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// logged in user
const edittedProfile = (user) => {
	return {
		type: "EDIT_PROFILE",
		payload: { user },
	};
};
