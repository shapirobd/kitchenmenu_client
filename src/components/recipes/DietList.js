import React from "react";
import EcoIcon from "@material-ui/icons/Eco";
import { useStyles } from "./styles/DietListStyles";
import {
	List,
	ListItem,
	ListItemText,
	ListItemAvatar,
	Avatar,
} from "@material-ui/core";

const DietList = ({ diets }) => {
	const classes = useStyles();

	return (
		<div className={classes.root}>
			<List>
				{diets.map((diet) => (
					<ListItem key={diet}>
						<ListItemAvatar>
							<Avatar className={classes.avatar}>
								<EcoIcon />
							</Avatar>
						</ListItemAvatar>
						<ListItemText primary={diet.toUpperCase()} />
					</ListItem>
				))}
			</List>
		</div>
	);
};

export default DietList;
