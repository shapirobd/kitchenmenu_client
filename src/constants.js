export const ALL_DIETS = [
	"glutenFree",
	"ketogenic",
	"vegetarian",
	"lactoVegetarian",
	"ovoVegetarian",
	"lactoOvoVegetarian",
	"vegan",
	"pescetarian",
	"paleo",
	"primal",
	"whole30",
];

export const ALL_MACROS = ["Fat", "Protein", "Carbohydrates"];

export const INITIAL_FILTER_DATA = {
	diets: [],
	macros: {
		Fat: {
			operator: null,
			amount: null,
		},
		Protein: {
			operator: null,
			amount: null,
		},
		Carbohydrates: {
			operator: null,
			amount: null,
		},
	},
};

export const WEEKDAYS = [
	"Sunday",
	"Monday",
	"Tuesday",
	"Wednesday",
	"Thursday",
	"Friday",
	"Saturday",
];

export const WEEKLY_COLORS = {
	carbs: [
		"#f44336",
		"#f44336",
		"#f44336",
		"#f44336",
		"#f44336",
		"#f44336",
		"#f44336",
	],
	fat: [
		"#4caf50",
		"#4caf50",
		"#4caf50",
		"#4caf50",
		"#4caf50",
		"#4caf50",
		"#4caf50",
	],
	protein: [
		"#2196f3",
		"#2196f3",
		"#2196f3",
		"#2196f3",
		"#2196f3",
		"#2196f3",
		"#2196f3",
	],
};
