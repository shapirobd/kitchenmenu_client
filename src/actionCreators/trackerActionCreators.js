import axios from "axios";
import { ADD_EATEN_MEAL, REMOVE_EATEN_MEAL } from "../components/actionTypes";
import { API_URL } from "../constants";

// const API_URL = "http://localhost:5000";

// adds eaten meal to database as well as redux state for a given date
// by dispatching action created by addedEatenMeal()
export const addEatenMeal = (username, recipeId, date, nutrients) => {
	return async (dispatch) => {
		try {
			const macros = {
				calories: Math.round(nutrients[0].amount),
				fat: Math.round(nutrients[1].amount),
				carbs: Math.round(nutrients[3].amount),
				protein: Math.round(nutrients[8].amount),
			};
			await axios.post(`${API_URL}/users/addEatenMeal`, {
				username,
				recipeId,
				date,
				nutrients: macros,
			});
			dispatch(addedEatenMeal(recipeId, date, macros));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe
// to be eaten and date that it was eaten
const addedEatenMeal = (recipeId, date, nutrients) => {
	const { calories, fat, carbs, protein } = nutrients;
	return {
		type: ADD_EATEN_MEAL,
		payload: {
			date,
			meal: {
				id: recipeId,
				calories,
				fat,
				carbs,
				protein,
			},
		},
	};
};

// removes eaten meal from database as well as redux state for a given date
// by dispatching action created by removedEatenMeal()
export const removeEatenMeal = (username, recipeId, date) => {
	return async (dispatch) => {
		try {
			await axios.post(`${API_URL}/users/removeEatenMeal`, {
				username,
				recipeId,
				date,
			});
			dispatch(removedEatenMeal(recipeId, date));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe
// to be removed and date that it was eaten
const removedEatenMeal = (recipeId, date) => {
	return {
		type: REMOVE_EATEN_MEAL,
		payload: {
			recipeId,
			date,
		},
	};
};
