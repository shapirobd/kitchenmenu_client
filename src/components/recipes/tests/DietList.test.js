import React from "react";
import { render, screen } from "@testing-library/react";
import DietList from "../DietList";
import { MemoryRouter } from "react-router-dom";
import { PersistGate } from "redux-persist/integration/react";
import { Provider } from "react-redux";
import { store, persistor } from "../../../configureStore";
import { login } from "../../../actionCreators/userActionCreators";
import { ALL_DIETS } from "../../../constants";

beforeAll(async () => {
	await store.dispatch(
		login({ username: "shapirobd", password: "Pot3ntiat321!" })
	);
});

it("should render without crashing", () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<DietList diets={ALL_DIETS} />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);
});

it("should match screenshot", () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<DietList diets={ALL_DIETS} />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);

	expect(asFragment()).toMatchSnapshot();
});
