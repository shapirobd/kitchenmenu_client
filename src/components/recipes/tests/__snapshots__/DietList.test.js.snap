// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-3"
  >
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            GLUTENFREE
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            KETOGENIC
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            VEGETARIAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            LACTOVEGETARIAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            OVOVEGETARIAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            LACTOOVOVEGETARIAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            VEGAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            PESCETARIAN
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            PALEO
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            PRIMAL
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.05 8.05c-2.73 2.73-2.73 7.15-.02 9.88 1.47-3.4 4.09-6.24 7.36-7.93-2.77 2.34-4.71 5.61-5.39 9.32 2.6 1.23 5.8.78 7.95-1.37C19.43 14.47 20 4 20 4S9.53 4.57 6.05 8.05z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            WHOLE30
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
